plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts

```diff
@@ -1,7 +1,13 @@
+export enum BlockOverlayAttribute {
+	NEVER = 'never',
+	MOBILE = 'mobile',
+	ALWAYS = 'always',
+}
```

Comment:
I guess you're using enum here to have a single source of truth for overlay option. But it also mean that TypeScript is not a superset of Javascript any more, as the value here is used in both type and runtime code. The code still works as expected, but I recommend using union type and constant instead of enum. You can still have a single source of truth by building the type from object, like this:

```ts
const OVERLAY = {
	never: 'never',
	mobile: 'mobile',
	always: 'always',
} as const;

type OverlayOptions = ( typeof OVERLAY )[ keyof typeof OVERLAY ]; // type OverlayOptions = "never" | "mobile" | "always"
```

Reply:
Thank you for your comment. I believe using `enum` is much more readable and easier to maintain the code than using typeof, keyof, etc. and two separate objects/types to handle that. But I know some part of the community prefers avoiding that, since I don't know what are the guidance regarding to this I changed the code in ee643ce8299135d0acedc10177e800f5316216ee